File: src/main/java/com/taskapp/model/Task.java
package com.taskapp.model;

import java.util.UUID;

public class Task {
    private String id;
    private String title;
    private String status; // "todo", "inprogress", "done"
    private long createdAt;

    public Task() {
        this.id = UUID.randomUUID().toString();
        this.createdAt = System.currentTimeMillis();
    }

    public Task(String title, String status) {
        this();
        this.title = title;
        this.status = status;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public long getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(long createdAt) {
        this.createdAt = createdAt;
    }
}

File: frontend/src/types/task.ts
export type TaskStatus = 'TODO' | 'IN_PROGRESS' | 'DONE';

export interface Task {
  id: number;
  title: string;
  description?: string;
  status: TaskStatus;
  priority?: number;
  dueDate?: string | null;
  createdBy?: string;
  updatedBy?: string | null;
  order?: number;
  createdAt: string;
  updatedAt: string;
}
